rust/agents/relayer/src/msg/metadata_builder.rs

```diff
@@ -43,38 +44,54 @@ impl MetadataBuilder {
         message: &HyperlaneMessage,
         mailbox: CachingMailbox,
     ) -> eyre::Result<Option<Vec<u8>>> {
+        const CTX: &str = "When fetching metadata";
+
         // The Mailbox's `recipientIsm` function will revert if
         // the recipient is not a contract. This can pose issues with
         // our use of the RetryingProvider, which will continuously retry
         // the eth_call to the `recipientIsm` function.
         // As a workaround, we avoid the call entirely if the recipient is
         // not a contract.
         let provider = mailbox.provider();
-        if !provider.is_contract(&message.recipient).await? {
+        if !provider
+            .is_contract(&message.recipient)
+            .await
+            .context(CTX)?
+        {
             info!(
                 recipient=?message.recipient,
                 "Could not fetch metadata: Recipient is not a contract"
             );
             return Ok(None);
         }
 
-        let ism_address = mailbox.recipient_ism(message.recipient).await?;
+        let ism_address = mailbox
+            .recipient_ism(message.recipient)
+            .await
+            .context(CTX)?;
         let multisig_ism = self
             .chain_setup
             .build_multisig_ism(ism_address, &self.metrics)
-            .await?;
+            .await
+            .context(CTX)?;
 
-        let (validators, threshold) = multisig_ism.validators_and_threshold(message).await?;
+        let (validators, threshold) = multisig_ism
+            .validators_and_threshold(message)
+            .await
+            .context(CTX)?;
         let highest_known_nonce = self.prover_sync.read().await.count() - 1;
-        let checkpoint_syncer = self.build_checkpoint_syncer(&validators).await?;
+        let checkpoint_syncer = self
+            .build_checkpoint_syncer(&validators)
```

Comment:
Again its been a bit for me in this code base, but I think in here we need most of the debug log lines?

Reply:
You said that you wanted more detailed errors specifically for the "Unable to reach quorum" case. The only way that info message is reached is if this line does not return an error. Following the path that can lead to this log took me to `checkpoint_syncer.fetch_checkpoint_in_range` retuning `Ok(None)`. Then looking at that function I added a bunch of deubg logs to the different conditions that can return `Ok(None)`.

Reply:
Let me know if I misunderstood your request.

Reply:
So think about the case when a validator does not announce themselves on `ValidatorAnnounce`. IIUC, `checkpoint_syncer` will not contain that validators s3 bucket, which will mean that `checkpoint_syncer.fetch_checkpoint_in_range` will not be able to reach quorum, resulting in the error message we are talking about. 

However, unless we add debug lines for `build_checkpoint_syncer`, an operator would not be able to determine this to be the source of their issues right?